### DISCUSSION TITLE
How to create ng template type web component using lit dev

### DISCUSSION THREAD

**Discussion by jitendraP-ashutec** [lit/lit#3383] [Q&A]
I want to create ng template type component Like:

```
<lit-web-component item="item">
    
    First Name: item.firstName
    Last Name: item.lastName

<lit-web-component>

```

My question is how we can parse it? In angular we can parse it like ${item.firstName}, but in web component how I can parse such of template ?  template content will be dynamic. Is there is approach so parse this template.

In simple work can I use string interpolation ? like ?

```
<lit-web-component item="item">
    
    First Name: ${item.firstName}
    Last Name: ${item.lastName}

<lit-web-component>
```

Any suggestion?

Thanks,
Jitendra Prajapati

**Comment by drk-mtr** [comment]
Are you referring to Angular's `<ng-template></ng-template>` (https://angular.io/api/core/ng-template), or something else?

**Reply by jitendraP-ashutec** [reply]
@drk-mtr  Yes, kind of ng template, I want user to add decide based condition display particular part, use repeat, and later user can change object value so based on that template value change.

**Comment by e111077** [comment]
Can you simply not make an lit template and just not render it? e.g.

https://lit.dev/playground/#gist=b343c3c7856d1b503aeaccdba64e7ea7

You can wrap this in a web component if you wish, but we highly advise AGAINST this pattern because you will generate a node even if it's not rendered. We used to do this in the Polymer library back in the day.

There is a proposal for native template instantiation as well:

https://github.com/WICG/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md

but in the meantime if you'd like that you might want to consider a library like Stampino:

https://www.npmjs.com/package/stampino/v/0.6.2

**Reply by jitendraP-ashutec** [reply]
Thanks @e111077  for your quick comment.  No I can't do like [this](https://lit.dev/playground/#gist=b343c3c7856d1b503aeaccdba64e7ea7). because I want allow to user to render based on some condition, repeat and do other thing based on user's requirement, later on based on user's input it will render it.


As I have explain in my question,  there will one object called **item** user will be pass and it will display item object property what ever he want, they may use condition, like in angular we have ngIf, ngFor. Does there is any package available in lit so we can allow such kind of functionality, so user can use that in any modern framework, like angular, react, vue ?

I have tried [https://www.npmjs.com/package/stampino/v/0.6.2](https://www.npmjs.com/package/stampino/v/0.6.2) in lit 
2.2.4 and in angular 14 , its not rendering. Its  didn't give any error but didn't render it.

Thanks,
Jitendra Prajapati
